### ISSUE TITLE
The Example for android can download bundle,but not applay

### ISSUE THREAD

**Issue by virtoolswebplayer** [Microsoft/react-native-code-push#153]
i test the example on ios and android ,
ios is ok,
but in android ,it failed, download bundle is ok  but not applay new update when reboot

**Comment by lostintangent** [comment]
Are you running in debug mode? On Android, React Native forces the JS bundle to be loaded from the packager, and therefore, doesn't ask the CodePush plugin where to get it from. So even after you install an update via CodePush, after the restart, it will still load the bundle from the packager and not from CodePush. We can't work around this.

Additionally, in debug mode, React Native caches the JS bundle, and therefore, even after you restart the app, it will re-use the locally available file unless you "force" it to reload. Once again, CodePush is completely out of control of this behavior.

Cpuld you try opening the dev menu and explicitly reload the bundle after you do an update? If that works, you can then enable "dev mode" for the CodePush plugin, which will automatically clear the JS bundle cache for you. You can do this by passing the "BuildConfig.DEBUG" property as the third parameter to the CodePush class's constructor in your MainActivity.java class.

**Comment by virtoolswebplayer** [comment]
check my code



#### test react-native  `0.17.0`  + react-native-code-push `1.5.2-beta`  on android

download is ok -> auto restart  -> alert me 'app can't running', -> manually restart the app -> app is up to date , but not apply update 


### mybundle.sh
```bash
#!/usr/bin/env bash

rm -rf ./release/*

react-native bundle --platform android  \
--entry-file index.android.js  \
--bundle-output ./release/index.android.bundle  \
--dev false

code-push release testandroid  ./release 1.0.0
```

### MainActivity.java
```java
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mReactRootView = new ReactRootView(this);

        codePush = new CodePush("this my key ", this, BuildConfig.DEBUG);

        mReactInstanceManager = ReactInstanceManager.builder()
                .setApplication(getApplication())
                .setJSBundleFile(codePush.getBundleUrl("index.android.bundle"))
                .addPackage(codePush.getReactPackage())

                .setJSMainModuleName("index.android")
                .addPackage(new MainReactPackage())

                //.setUseDeveloperSupport(BuildConfig.DEBUG)
                .setInitialLifecycleState(LifecycleState.RESUMED)
                .build();

        mReactRootView.startReactApplication(mReactInstanceManager, "codepush", null);

        setContentView(mReactRootView);
    }

```
###  crossplatformdemo.js
```javascript
'use strict';

import React, {
  AppRegistry,
  Dimensions,
  Image,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import Button from "react-native-button";
import CodePush from "react-native-code-push";

let CodePushDemoApp = React.createClass({
  async sync() {
    let self = this;
    try {
      return await CodePush.sync(
        { 
          updateDialog: true,
          installMode: CodePush.InstallMode.IMMEDIATE
        }, 
        (syncStatus) => {
          switch(syncStatus) {
            case CodePush.SyncStatus.CHECKING_FOR_UPDATE: 
              self.setState({
                syncMessage: "Checking for update."
              });
              break;
            case CodePush.SyncStatus.DOWNLOADING_PACKAGE:
              self.setState({
                syncMessage: "Downloading package."
              });
              break;
            case CodePush.SyncStatus.AWAITING_USER_ACTION:
              self.setState({
                syncMessage: "Awaiting user action."
              });
              break;
            case CodePush.SyncStatus.INSTALLING_UPDATE:
              self.setState({
                syncMessage: "Installing update."
              });
              break;
            case CodePush.SyncStatus.UP_TO_DATE:
              self.setState({
                syncMessage: "App up to date.",
                progress: false
              });
              break;
            case CodePush.SyncStatus.UPDATE_IGNORED:
              self.setState({
                syncMessage: "Update cancelled by user.",
                progress: false
              });
              break;
            case CodePush.SyncStatus.UPDATE_INSTALLED:
              self.setState({
                syncMessage: "Update installed and will be run when the app next resumes.",
                progress: false
              });
              break;
            case CodePush.SyncStatus.UNKNOWN_ERROR:
              self.setState({
                syncMessage: "An unknown error occurred.",
                progress: false
              });
              break;
          }
        },
        (progress) => {
          self.setState({
            progress: progress
          });
        }
      );
    } catch (error) {
      CodePush.log(error);
    }
  },
  
  getInitialState() {
    return { };
  },
  
  render() {
    let syncView, syncButton, progressView;
    
    if (this.state.syncMessage) {
      syncView = (
        <Text style={styles.messages}>{this.state.syncMessage}</Text>
      );
    } else {
      syncButton = ( 
        <Button style={{color: 'green'}} onPress={this.sync}>
          Start Sync!
        </Button>
      );
    }
    
    if (this.state.progress) {
      progressView = (
        <Text style={styles.messages}>{this.state.progress.receivedBytes} of {this.state.progress.totalBytes} bytes received</Text>
      );
    }
    
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          ios and anroid version 2.7
        </Text>
        {syncButton}
        {syncView}
        {progressView}
      </View>
    );
  }
});

let styles = StyleSheet.create({
  image: {
    marginTop: 50,
    width: Dimensions.get('window').width - 100,
    height: 365 * (Dimensions.get('window').width - 100) / 651,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    paddingTop: 50
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  },
  messages: {
    textAlign: 'center',
  },
});

AppRegistry.registerComponent('codepush', () => CodePushDemoApp);

```

**Comment by lostintangent** [comment]
On Android, we currently only support releasing the JS bundle file directly. Could you modify your release script to specify "code-push release testandroid ./release/index.android.bundle 1.0.0" instead of uploading the entire "release" folder? This should fix your issue, as I believe the plugin is failing when it would try to install the update, since it was expecting a JS bundle file, but was getting a zip file.

Let me know if that fixes your problem. Thanks so much for the awesome details for the repro!

**Comment by virtoolswebplayer** [comment]
thanks a lot ' it's ok !!  please close this issues

**Comment by lostintangent** [comment]
Yay! Thanks for confirming and I'm glad you got it working. Let us know if you run into any other issues.

**Comment by kamek-pf** [comment]
Hey, sorry to bump this @lostintangent , but :
> On Android, React Native forces the JS bundle to be loaded from the packager

I also ran into this, and it's quite confusing that the updates aren't applied as you'd expect when in development mode. 

I spent a fair part of the day trying to understand what I did wrong. Turns out everything was set up correctly, but it took me a while to figure it out.
I feel like this should be mentioned in the docs.

Other than that, very nice tool, keep up the good work !
